### DISCUSSION TITLE
GraphQL API : linking issue to PR and updatePullRequest bug with projectIds

### DISCUSSION THREAD

**Discussion by MetMS** [community/community#76626] [Apps, API and Webhooks]
### Select Topic Area

Bug

Hello,

I'm having two issues and I would like to know if other persons have the same.

**First one** is how to link an issue (not a pull request) to a pull request by using GraphQL API. I've been looking in the documentation but with CreatePullRequest or UpdatePullRequest I don't see input allowing that.

I would like to know if someone know another way to do that ?

**Second one** is that with UpdatePullRequest I have the following error  (value is xxxx for security):
```json
{"data":{"updatePullRequest":null},"errors":[{"type":"NOT_FOUND","path":["updatePullRequest"],"locations":[{"line":2,"column":9}],"message":"Could not resolve to P
roject node with the global id of 'PVT_xxxxxxxxx-8'"}]}
```
I've added the header 'X-Github-Next-Global-ID': 1 but it doesn't do anything. Even with Github Explorer i'm having this issue.

By the way, the id of project come from this query :

```graphql
query {
      repository(owner:"xxxxxxxxxx", name: "xxxxxxxxxx") {
        pullRequest(number:xxxxxxxxx){
          title,
          body,
          assignees(first:1){
          nodes{
                  id
              }
          }
          url
          projectsV2(first:1){
            nodes{
              id
            }
          }
          milestone{
            id
          }
          closingIssuesReferences(first:1){
            nodes{
              id
            }
          }
        }
      }
    }
```
Thanks in advance for any help :)

**Comment by Naimulnobel** [comment]
1. **Linking an issue to a pull request**: Currently, the GitHub GraphQL API does not support linking an issue to a pull request directly. However, you can indirectly link an issue to a pull request by mentioning the issue in the pull request description or in a commit message. For example, you can write "Closes #issue_number" in the pull request description, and GitHub will automatically link the issue to the pull request.

2. **Error with UpdatePullRequest**: The error message suggests that the ID you're providing does not correspond to a valid Project node. The ID should be the global ID of the pull request you want to update, not the project ID. You can get the pull request ID from the `pullRequest` field in your query. Make sure you're using the correct ID in your `UpdatePullRequest` mutation.

Regarding the `X-Github-Next-Global-ID` header, this is not a standard GitHub API header and it's unclear what it's supposed to do. It's not mentioned in the GitHub API documentation, so it's recommended to remove it.

If you're still having issues, please provide more details about your `UpdatePullRequest` mutation and the exact error message you're seeing.
